### PR TITLE
add fips upload schemas to ohi.yml

### PR DESCRIPTION
**Pull request by rajrohanyadav**
We need to publish FIPS packages in addition to the existing packages. Therefore, update the schema for s3 upload.

[S3 after prerelease run with this schema for nri-redis](http://nr-downloads-ohai-staging.s3-website-us-east-1.amazonaws.com/rohan-fips-test-redis/)

***Note:*** ~This is a breaking change until all OHIs are updated to have FIPS packages.~

### CODE REVIEW

**Comment by sigilioso** [comment]
> Note: This is a breaking change until all OHIs are updated to have FIPS packages.

🤔  This is a bit challenging, we could add `v2` tag after merging but, in order to apply it gradually (when each OHI is ready) we will also need to consider the bump to to `infrastructure-publish-action@v2` in [coreint-automation](https://github.com/newrelic/coreint-automation/tree/main) a breaking change and also bump the _major_ there. After that, we could bump the reusable-workflows usage when each OHI is ready.

**Comment by rajrohanyadav** [comment]
> > Note: This is a breaking change until all OHIs are updated to have FIPS packages.
> 
> 🤔 This is a bit challenging, we could add `v2` tag after merging but, in order to apply it gradually (when each OHI is ready) we will also need to consider the bump to to `infrastructure-publish-action@v2` in [coreint-automation](https://github.com/newrelic/coreint-automation/tree/main) a breaking change and also bump the _major_ there. After that, we could bump the reusable-workflows usage when each OHI is ready.

Yes, you are correct. One thing that I explored was to create a new schema file like `ohi-fips.yaml` and pass that as an input in the [coreint-automation](https://github.com/newrelic/coreint-automation/tree/main) code. But that would involve merging these duplicated files and redoing some work. I thought updating the tags made sense. But open for inputs.

File: schemas/ohi.yml [hunk 1]
```diff
@@ -59,6 +67,31 @@
         - bullseye
         - bookworm
 
+- src: "{app_name}_{version}-1_{arch}_fips.deb"
```
Comment (rubenruizdegauna) [inline schemas/ohi.yml:70]: I wonder if the `fips` should be part of the `app_name` and not a suffix.

i.e. app_name=`newrelic-infra-fips`

https://www.debian.org/doc/manuals/debian-reference/ch02.en.html#_debian_package_file_names

Same for `rpm`. Have you already discussed this?

Comment (rajrohanyadav) [reply schemas/ohi.yml:70]: No, this has not been discussed. But I agree with you. `fips` in the name before `arch` makes sense.

Comment (alvarocabanas) [reply schemas/ohi.yml:70]: I'll tell Dan also to do the change.

Comment (alvarocabanas) [reply schemas/ohi.yml]: Notified for the infra-agent PR in here https://github.com/newrelic/infrastructure-agent/pull/1942#pullrequestreview-2451394416

Comment (rajrohanyadav) [reply schemas/ohi.yml]: Done, thanks for pointing out.

**Comment by rajrohanyadav** [comment]
> > Note: This is a breaking change until all OHIs are updated to have FIPS packages.
> 
> 🤔 This is a bit challenging, we could add `v2` tag after merging but, in order to apply it gradually (when each OHI is ready) we will also need to consider the bump to to `infrastructure-publish-action@v2` in [coreint-automation](https://github.com/newrelic/coreint-automation/tree/main) a breaking change and also bump the _major_ there. After that, we could bump the reusable-workflows usage when each OHI is ready.

As it turns out, to minimize the changes required, it's better if we have two files for fips and non-fips. This is because `app_name` needs to reflect whether it is fips or not for metadata updates to the repos. So, I'm planning to add a new task in [coreint-automation](https://github.com/newrelic/coreint-automation/compare/main...test-fips-s3-upload-schema) which will upload fips packages with the fips app_name.

**Review by alvarocabanas — APPROVED** [review]
LGTM

**Pull request merged**